adam/perception/high_level_semantics_situation_to_developmental_primitive_perception.py

```diff
@@ -278,6 +297,28 @@ def _get_relations_from_condition(
 
         return relations
 
+    def _get_manipulator_of_agent(self, agent_object: SituationObject) -> OntologyNode:
```

Comment:
I think we want something a bit more general here.  For example, if we fail to find the explicit binding for a slot filler to an argument role, we want to then search over all unbound objects in the scene to find one with the required properties.

Reply:
I think we already add sub-objects to the scene before we deal with relations, right?

Reply:
Didn't realize that, working on it now. How should we handle cases where there are multiple objects that fit the property? For example, for "give", we want to associate the manipulator in the precondition with the agent and manipulator in the postcondition with the goal.

Reply:
For now, throw an exception if we can't unambiguously resolve a variable to an object.

I am not sure I understand "manipulator in the postcondition with the goal." Can you explain?

Reply:
That works. For example A person can give a ball to another person. In the post condition, the second person would hold the ball with their manipulator, while there are multiple manipulators in the situation.